### PR TITLE
fix(sash): added typing and utilised TextProps

### PR DESCRIPTION
**Pull request by darmaine-qh**
## Description

* Fix to add typing for `<Sash />` for Typescript consumers
* Enhanced props by utilising `TextProps` applied in `Icon` and `BadgeText`
💁‍

## Related issues
n/a - bug fix

💪

## Looks like this
Screen Shot 2021-09-01 at 12 37 15 pm

🤨

---

**PR check-list**

🚨 **IMPORTANT:** Please review the [guidelines for contributing](./CONTRIBUTING.md) to this repository first! 👀

- [x] You've requested a review from at least ONE person. Preferably a roo-ui maintainer ([Philip](https://github.com/philipwindeyer), [Eric](https://github.com/eneo5541), [Mike](https://github.com/memcc), [Marcel](https://github.com/Anthropic), [Sam](https://github.com/SamuSan) [Long](https://github.com/KieraDOG)) or a Roo UI working-group human.
  - Note: If your PR has not been reviewed within two days of requesting a review, feel free to seek reviews from other [team members of Qantas Hotels](https://github.com/orgs/hooroo/people).
- [ ] Your final merge commit message is in the format `<type>(<scope>): <description>` as per [the contributing guide](https://github.com/hooroo/roo-ui/blob/master/.github/CONTRIBUTING.md#commit-naming)
  - Example: `fix(dependencies): minor version bump to resolve security vulnerability in set-value`

### CODE REVIEW

**Comment by ehyland** [comment]
👷 Deploy Preview for *roo-ui* processing.


🔨 Explore the source changes: 45ce08e7ec021f231de093d70aff2a972caf22e0

🔍 Inspect the deploy log: [https://app.netlify.com/sites/roo-ui/deploys/612ee789b6e7cb0007c042ca](https://app.netlify.com/sites/roo-ui/deploys/612ee789b6e7cb0007c042ca?utm_source=github&utm_campaign=bot_dl)

**Review by eneo5541 — APPROVED** [review]

**Review by philipwindeyer — APPROVED** [review]
🙌

File: src/components/Sash/Sash.js [hunk 1]
```diff
@@ -18,37 +18,33 @@ const Badge = styled(Box)`
...
+
 Sash.propTypes = {
   text: PropTypes.string,
-  fontSize: PropTypes.oneOfType([
-    PropTypes.number,
-    PropTypes.string,
-    PropTypes.array,
-  ]),
   iconName: PropTypes.string,
+  ...Text.defaultProps,
```
Comment (mrm007) [inline src/components/Sash/Sash.js:41]: Did you mean to spread `Text.propTypes`?

**Pull request merged**